File: ChatGPT4/D4/insert_sort.c
#include <stdio.h>

int arr[1000];

void swap(int *a, int *b) {
    int tmp = *a;
    *a = *b;
    *b = tmp;
}

void print_all(int MAX) {
    for(int i=0; i<MAX; i++)
        printf("%d ", arr[i]);
    printf("\n");
}

void input_data(int MAX) {
    int i=0;
    while(i < MAX) {
        scanf("%d", &arr[i++]);
    }
}
int main() {
    int MAX;
    scanf("%d", &MAX);

    input_data(MAX);
    
    int i = 0, j;
    for(i=1; i<MAX; i++) {
        for(j=i; j>0; j--) {
            //printf("%d %d %d\n", i, j, j-1);
            if(arr[j]<arr[j-1]) {
                int tmp = arr[j];
                arr[j] = arr[j-1];
                arr[j-1] = tmp;
            }
        }
    }
    print_all(MAX);
    return 0;
}

File: ChatGPT4/D4/insert_sort.py
nums = [11,22,44,33,77,3,55,66,88,99,0]

for i in range(1, len(nums)):
    for j in range(i, 0, -1):
        #print(i, j, j - 1)
        if(nums[j-1] > nums[i]):
            tmp = nums[j-1]
            nums[j-1] = nums[i]
            nums[i] = tmp
    print(i, nums)